beat/plotting.py

```diff
@@ -1848,42 +1853,92 @@ def fuzzy_spectra(axes, data, best_result, bbox_to_anchor, lower_corner=0.0,
         in_ax.xaxis.set_visible(False)
         in_ax.spines['bottom'].set_visible(False)
         
-        for tmark, ybound in zip([fxdata[lower], fxdata[upper-1]], [0.5*ymax, ymax]):
-            in_ax.plot(
-                [tmark, tmark], [0.0, ybound], color=tap_color_annot, lw=0.75)
-        
-        # annotate axis amplitude
-        in_ax.annotate(
-            '%0.3g -' % (ymax),
-            xycoords='data',
-            xy=(fxdata[upper-1], 0.9*ymax),
-            xytext=(1., 1.),
-            textcoords='offset points',
-            ha='right',
-            va='center',
-            fontsize=fontsize - 3,
-            color=obs_color,
-            fontstyle='normal')
+        if allaxes:
+
+            for tmark, ybound in zip([fxdata[lower], fxdata[upper-1]], [0.6*ymax, 0.6*ymax]):
+                in_ax.plot(
+                    [tmark, tmark], [0.0, ybound], color=tap_color_annot, lw=0.75)
+
+            # annotate axis amplitude
+            in_ax.annotate(
+                '%0.3g -' % (ymax),
+                xycoords='data',
+                xy=(fxdata[upper-1], 0.45*ymax),
+                xytext=(1., 1.),
+                textcoords='offset points',
+                ha='right',
+                va='center',
+                fontsize=fontsize - 3,
+                color=obs_color,
+                fontstyle='normal')
+    
+            in_ax.annotate(
+                '$ f \ |\ ^{%0.1g}_{%0.1g} \ $' % (fxdata[lower], fxdata[upper]),
+                xycoords='data',
+                xy=(fxdata[upper-1], 0.2*ymax),
+                xytext=(1., 1.),
+                textcoords='offset points',
+                ha='right',
+                va='center',
+                fontsize=fontsize+1,
+                color=obs_color,
+                fontstyle='normal')
+
+        else:
+
+            for tmark, ybound in zip([fxdata[lower], fxdata[upper-1]], [0.5*ymax, ymax]):
+                in_ax.plot(
+                    [tmark, tmark], [0.0, ybound], color=tap_color_annot, lw=0.75)
+            # annotate axis amplitude
+            in_ax.annotate(
+                '%0.3g -' % (ymax),
+                xycoords='data',
+                xy=(fxdata[upper-1], 0.9*ymax),
+                xytext=(1., 1.),
+                textcoords='offset points',
+                ha='right',
+                va='center',
+                fontsize=fontsize - 3,
+                color=obs_color,
+                fontstyle='normal')
+    
+            in_ax.annotate(
+                '$ f \ |\ ^{%0.1g}_{%0.1g} \ $' % (fxdata[lower], fxdata[upper]),
+                xycoords='data',
+                xy=(fxdata[upper-1], 0.4*ymax),
+                xytext=(1., 1.),
+                textcoords='offset points',
+                ha='right',
+                va='center',
+                fontsize=fontsize+1,
+                color=obs_color,
+                fontstyle='normal')
+    
+    def refining_targets(targets, spcindices, tmtrgts):
```

Comment:
an idea would be to introduce a SpectralTarget class that would be used in the wavemaps instead of the DynamicTarget- you would inherit from it and could put spectra-related stuff there. book-keeping would become much easier
also this hacking with the nslc_id would become not necessary this would right away be done by the spectral target